scripts/auto_mesh.py

```diff
@@ -0,0 +1,290 @@
+        entities = SimpleParser(self.filename).parse()
+        for name, properties in entities:
+            if name == "listener" and properties and properties.get("role") == "inter-router":
+                outval = {"port":properties.get("port", "55672")}
+                add_sasl_properties(outval)
+                if "sslProfile" in properties:
+                    outval["sslProfile"] = properties["sslProfile"]
+                    outval["verifyHostName"] = "no"
+                return outval
+        return {"port":"55672"}
+
+    def append_connectors(self, connectors):
+        properties = self.get_inter_router_properties()
+        with open(self.filename, "a") as f:
+            print("", file=f)
+            print("# generated by auto-mesh %s" % mode, file=f)
+            for c in connectors:
+                c.update(properties)
+                print("connector {", file=f)
+                for k,v in c.items():
+                    print("    %s: %s" % (k, v), file=f)
+                print("}", file=f)
+
+def get_config(config_file):
+    try:
+        return JsonConfig(config_file)
+    except:
+        return SimpleConfig(config_file)
+
+def query():
+    host = os.environ['KUBERNETES_SERVICE_HOST']
+    port = os.environ['KUBERNETES_PORT_443_TCP_PORT']
+    ip   = os.environ['POD_IP']
+    ns   = os.environ['POD_NAMESPACE']
+    name = os.environ['APPLICATION_NAME']
+
+    path = "/api/v1/namespaces/" + ns + "/pods"
+    token = retrieve_token()
+
+    while True:
+        # Send REST requests until other routers are ready
+        response = api_request(host, port, path, token)
+        ip_list  = extract_ips(response, name)
+        if ip_list is not None:
+            break
+        # ip_list is empty, try again after one second
+        time.sleep(1)
+
+    # Get the ip addresses of the routers already running
+    if ip_list:
+        # Keep only pods created after current pod
+        si = ip_list.index(ip)
+        ip_list = ip_list[:si]
+        return [{"role": "inter-router", "host":host} for host in ip_list]
+    return ip_list
+
+def infer():
+    service_name = os.environ.get("QDROUTERD_AUTO_MESH_SERVICE_NAME", "%s-headless" % os.environ.get("APPLICATION_NAME", "amq-interconnect"))
+    namespace = retrieve_namespace()
+    (prefix, index) = os.environ["HOSTNAME"].rsplit("-", 1)
+    return [{"role": "inter-router", "host":"%s-%s.%s.%s.svc.cluster.local" % (prefix, i, service_name, namespace)} for i in range(int(index))]
+
+if __name__ == "__main__":
+    mode = os.environ.get("QDROUTERD_AUTO_MESH_DISCOVERY")
+    if mode and len(sys.argv) > 1:
+        try:
+            if mode.upper() == "QUERY":
+                connectors = query()
+            elif mode.upper() == "INFER":
+                connectors = infer()
+            else:
+                raise Exception("Invalid value for QDROUTERD_AUTO_MESH_DISCOVERY, expected 'QUERY' or 'INFER'")
+            get_config(sys.argv[1]).append_connectors(connectors)
```

Comment:
Should this feature be (eventually) moved into skuperrouter itself?

Reply:
It is possible. After the router is started, Skupper already maintains connectors properly.
However these scripts are used only to prepare the config for the router startup, before qdrouterd is executed.

Reply:
Since skupper-router is going to be exclusively in skupper, then making the option 'QDROUTERD_AUTO_MESH_DISCOVERY' be qdrouterd's option sounds sensible to me. It's an enhancement idea, however; not relevant for this PR right now.

Reply:
This is certainly something we can discuss and re-evaluate.

Reply:
follow up issue https://github.com/skupperproject/skupper-router/issues/60